### PR TITLE
[wasm] mount VFS archive instead of copying it to MEMFS

### PR DESCRIPTION
**Pull request by pavelsavara**
- reduces load of `dotnet.timezones.blat` from 60ms to 3ms
- remove `free` on method name which should not be done

@maraf this doesn't really solve it for memory snapshot. But I think we could implement similar read only VFS overlay in `libs/native` level and solve this also for WASI. Storing the files and hierarchy directly in the main memory.

### CODE REVIEW

**Comment by pavelsavara** [comment]
`System.Tests.TimeZoneInfoTests.IsDaylightSavingTime`
`Could not find a part of the path '/usr/share/zoneinfo/Europe/London'.`

File: src/mono/wasm/runtime/assets.ts [hunk 1]
```diff
@@ -427,6 +427,85 @@ function _instantiate_asset(asset: AssetEntry, url: string, bytes: Uint8Array) {
...
+            llseek: noop
+        }
+    };
+    FS.mkdirTree(prefix);
+    const root = FS.mount(ARCHIVEFS, { root: "." }, prefix);
+
+    let offset = 0;
+    for (const row of manifest) {
+        const name = row[0];
+        const length = row[1];
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/assets.ts:492]: Row[1] == length of the path? How do we know it? Is the manifest a file that I could peek into?

File: src/mono/wasm/runtime/assets.ts [hunk 2]
```diff
@@ -482,31 +561,7 @@ export function mono_wasm_load_data_archive(data: Uint8Array, prefix: string): b
...
-    });
-    folders.forEach(folder => {
-        Module["FS_createPath"](prefix, folder, true, true);
-    });
-
-    for (const row of manifest) {
-        const name = row[0];
-        const length = row[1];
-        const bytes = data.slice(0, length);
-        Module["FS_createDataFile"](prefix, name, bytes, true, true);
```
Comment (ilonatommy) [inline src/mono/wasm/runtime/assets.ts:507]: Here we are writing data to the file we created. Why then are we creating nodes in FS in the new approach with read-only-mode?

Comment (pavelsavara) [reply src/mono/wasm/runtime/assets.ts:507]: Why would the application try to write into timezone data ? Making it RW would be more code and complexity.

File: src/mono/wasm/runtime/assets.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/assets.ts:492]: The manifest never hits the file system in JSON form. See `src\tasks\WasmBuildTasks\GenerateWasmBundle.cs` how it's created. If you open `artifacts\bin\microsoft.netcore.app.runtime.browser-wasm\Debug\runtimes\browser-wasm\native\dotnet.timezones.blat` it's pretty much readable.

**Pull request closed without merging**